Rename a representation/document with a name length > 50 characters is not properly handled

* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- latest
* [x] **I have specified my environment** : 
- All

### Stack trace

Server stack trace after the rename:

```
2023-03-01 09:37:39.625 ERROR 98777 --- [14-f65b9545938c] o.h.engine.jdbc.spi.SqlExceptionHelper   : ERROR: new row for relation "representation" violates check constraint "representation_label_length"
  Détail : Failing row contains (df0bc3ba-3685-4a04-8ddf-2467f05262ff, 46e24162-b280-4843-af14-f65b9545938c, 4518b809-169c-473a-ac7a-b0a7ed36cb5a, Averyveryveryverylongnamewithoutspacesazertyuiopaz, siriusComponents://representation?type=Diagram, 29228857-eb70-346d-ba1f-b6844b21a977, {"id":"df0bc3ba-3685-4a04-8ddf-2467f05262ff","kind":"siriusCompo...).
2023-03-01 09:37:39.627  INFO 98777 --- [14-f65b9545938c] o.h.e.j.b.internal.AbstractBatchImpl     : HHH000010: On release of batch it still contained JDBC statements
2023-03-01 09:37:39.653  WARN 98777 --- [14-f65b9545938c] o.e.s.c.c.e.EditingContextEventProcessor : could not execute statement; SQL [n/a]; constraint [representation_label_length]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement

org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [representation_label_length]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:276) ~[spring-orm-5.3.23.jar:5.3.23]
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:233) ~[spring-orm-5.3.23.jar:5.3.23]
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:566) ~[spring-orm-
	at ...
	at org.eclipse.sirius.web.services.representations.RepresentationService.save(RepresentationService.java:123) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.diagrams.handlers.RenameDiagramEventHandler.handle(RenameDiagramEventHandler.java:91) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.diagrams.DiagramEventProcessor.handle(DiagramEventProcessor.java:119) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.editingcontext.EditingContextEventProcessor.handleRepresentationInput(EditingContextEventProcessor.java:325) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.editingcontext.EditingContextEventProcessor.doHandle(EditingContextEventProcessor.java:245) ~[classes/:na]
	at ... org.eclipse.sirius.components.collaborative.trees.handlers.RenameTreeItemEventHandler.handle(RenameTreeItemEventHandler.java:110) ~[classes/:na]
	at org.eclipse.sirius.components.collaborative.trees.TreeEventProcessor.handle(TreeEventProcessor.java:118) ~[classes/:na]
	....
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:109) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:37) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:200) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.batch.internal.NonBatchingBatch.addToBatch(NonBatchingBatch.java:46) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3571) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.updateOrInsert(AbstractEntityPersister.java:3438) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3852) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.action.internal.EntityUpdateAction.execute(EntityUpdateAction.java:202) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:604) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.spi.ActionQueue.lambda$executeActions$1(ActionQueue.java:478) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[na:na]
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:475) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:344) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:40) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1407) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:489) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:3303) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2438) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:449) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:183) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$300(JdbcResourceLocalTransactionCoordinatorImpl.java:40) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:281) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101) ~[hibernate-core-5.6.12.Final.jar:5.6.12.Final]
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:562) ~[spring-orm-5.3.23.jar:5.3.23]
	... 36 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: new row for relation "representation" violates check constraint "representation_label_length"
  Détail : Failing row contains (df0bc3ba-3685-4a04-8ddf-2467f05262ff, 46e24162-b280-4843-af14-f65b9545938c, 4518b809-169c-473a-ac7a-b0a7ed36cb5a, Averyveryveryverylongnamewithoutspacesazertyuiopaz, siriusComponents://representation?type=Diagram, 29228857-eb70-346d-ba1f-b6844b21a977, {"id":"df0bc3ba-3685-4a04-8ddf-2467f05262ff","kind":"siriusCompo...).
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675) ~[postgresql-42.3.7.jar:42.3.7]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2365) ~[postgresql-42.3.7.jar:42.3.7]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:355) ~[postgresql-42.3.7.jar:42.3.7]
	... 58 common frames omitted
```

### Steps to reproduce

1. Use the Robot Flow model
2. Select the System item in the Explorer View
3. Create a new representation on the system element with name > 50 characters

### Expected behavior

The new name is taken into account or is not taken into account but an error message appears in the snackbar.
It is possible to rename any other tree item or any field in the details view after that.

### Actual behavior

The new name is not taken into account.
It is impossible to rename any other tree item or any field in the details view after that.

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -43,6 +43,7 @@
 For that, specifiers will be able to use properly the variable name property of the create view model operation.
 The result returned and captured in this variable name will not be the real node since it will be rendered later but it is a placeholder which contains some properties of the real node such as its id which can be used to created the nested views.
 - https://github.com/eclipse-sirius/sirius-components/issues/1838[#1838] [diagram] Collapsing and expanding a node with unsynchronized child nodes will now properly restore the child nodes
+- https://github.com/eclipse-sirius/sirius-components/issues/1775[#1775] [core] Handle representation/document/project label length in UI (by adding ellipsis) and in sirius-web SQL schema (up to 1024 characters)
 
 === New Features
 

--- a/packages/core/frontend/sirius-components/src/modals/rename-project/RenameProjectModalMachine.ts
+++ b/packages/core/frontend/sirius-components/src/modals/rename-project/RenameProjectModalMachine.ts
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2021, 2022 Obeo.
+ * Copyright (c) 2021, 2023 Obeo.
  * This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v2.0
  * which accompanies this distribution, and is available at
@@ -58,7 +58,7 @@ export type RenameProjectModalEvent =
   | RequestProjectRenamingEvent
   | HandleResponseEvent;
 
-const isNameInvalid = (name: string) => name.trim().length < 3 || name.trim().length > 20;
+const isNameInvalid = (name: string) => name.trim().length < 3 || name.trim().length > 1024;
 export const renameProjectModalMachine = Machine<
   RenameProjectModalContext,
   RenameProjectModalStateSchema,
@@ -68,7 +68,7 @@ export const renameProjectModalMachine = Machine<
     type: 'parallel',
     context: {
       name: null,
-      nameMessage: 'The name must contain between 3 and 20 characters',
+      nameMessage: 'The name must contain between 3 and 1024 characters',
       nameIsInvalid: false,
       initialName: null,
       message: null,

--- a/packages/forms/frontend/sirius-components-forms/src/propertysections/ListPropertySection.tsx
+++ b/packages/forms/frontend/sirius-components-forms/src/propertysections/ListPropertySection.tsx
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2019, 2022 Obeo.
+ * Copyright (c) 2019, 2023 Obeo.
  * This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v2.0
  * which accompanies this distribution, and is available at
@@ -62,6 +62,10 @@ export const clickListItemMutation = gql`
 `;
 
 const useListPropertySectionStyles = makeStyles<Theme, ListStyleProps>((theme) => ({
+  table: {
+    tableLayout: 'fixed',
+    width: '100%',
+  },
   cell: {
     display: 'flex',
     flexDirection: 'row',
@@ -84,6 +88,10 @@ const useListPropertySectionStyles = makeStyles<Theme, ListStyleProps>((theme) =
     fontStyle: ({ italic }) => (italic ? 'italic' : 'inherit'),
     fontWeight: ({ bold }) => (bold ? 'bold' : 'inherit'),
     textDecorationLine: ({ underline, strikeThrough }) => getTextDecorationLineValue(underline, strikeThrough),
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    flexGrow: 1,
   },
 }));
 
@@ -235,36 +243,36 @@ export const ListPropertySection = ({
 
   const getTableCellContent = (item: GQLListItem): JSX.Element => {
     return (
-      <Typography
-        className={`${classes.canBeSelectedItem} ${classes.style}`}
-        onClick={() => clickHandler(item)}
-        color="textPrimary"
-        data-testid={`representation-${item.id}`}>
+      <>
         {item.imageURL ? (
-          <img className={classes.icon} width="16" height="16" alt={item.label} src={httpOrigin + item.imageURL} />
+          <img className={classes.icon} width="16" height="16" alt={''} src={httpOrigin + item.imageURL} />
         ) : null}
-        {item.label}
-      </Typography>
+        <Typography
+          className={`${classes.canBeSelectedItem} ${classes.style}`}
+          onClick={() => clickHandler(item)}
+          color="textPrimary"
+          data-testid={`representation-${item.id}`}>
+          {item.label}
+        </Typography>
+        <IconButton
+          aria-label="deleteListItem"
+          onClick={(event) => onDelete(event, item)}
+          disabled={readOnly || !item.deletable}
+          data-testid={`delete-representation-${item.id}`}>
+          <DeleteIcon />
+        </IconButton>
+      </>
     );
   };
 
   return (
     <FormControl error={widget.diagnostics.length > 0} fullWidth>
       <PropertySectionLabel label={widget.label} subscribers={subscribers} />
-      <Table size="small">
+      <Table size="small" className={classes.table}>
         <TableBody>
           {items.map((item) => (
             <TableRow key={item.id}>
-              <TableCell>{getTableCellContent(item)}</TableCell>
-              <TableCell align="right">
-                <IconButton
-                  aria-label="deleteListItem"
-                  onClick={(event) => onDelete(event, item)}
-                  disabled={readOnly || !item.deletable}
-                  data-testid={`delete-representation-${item.id}`}>
-                  <DeleteIcon />
-                </IconButton>
-              </TableCell>
+              <TableCell className={classes.cell}>{getTableCellContent(item)}</TableCell>
             </TableRow>
           ))}
         </TableBody>

--- a/packages/sirius-web/backend/sirius-web-persistence/src/main/resources/db/changelog/sirius-web-schema.sql
+++ b/packages/sirius-web/backend/sirius-web-persistence/src/main/resources/db/changelog/sirius-web-schema.sql
@@ -5,7 +5,7 @@ CREATE TABLE document (
     project_id uuid NOT NULL,
     name text NOT NULL,
     content text NOT NULL,
-    CONSTRAINT document_name_length CHECK (((char_length(name) > 0) AND (char_length(name) <= 50)))
+    CONSTRAINT document_name_length CHECK (((char_length(name) > 0) AND (char_length(name) <= 1024)))
 );
 
 CREATE TABLE idmapping (
@@ -16,7 +16,7 @@ CREATE TABLE idmapping (
 CREATE TABLE project (
     id uuid DEFAULT gen_random_uuid() NOT NULL,
     name text NOT NULL,
-    CONSTRAINT project_name_length CHECK (((char_length(name) > 0) AND (char_length(name) <= 50)))
+    CONSTRAINT project_name_length CHECK (((char_length(name) > 0) AND (char_length(name) <= 1024)))
 );
 
 CREATE TABLE representation (
@@ -27,7 +27,7 @@ CREATE TABLE representation (
     kind text NOT NULL,
     descriptionId TEXT NOT NULL,
     content text NOT NULL,
-    CONSTRAINT representation_label_length CHECK (((char_length(label) > 0) AND (char_length(label) <= 50)))
+    CONSTRAINT representation_label_length CHECK (((char_length(label) > 0) AND (char_length(label) <= 1024)))
 );
 
 ALTER TABLE ONLY idmapping

--- a/packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/projects/ProjectService.java
+++ b/packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/projects/ProjectService.java
@@ -146,7 +146,7 @@ public class ProjectService implements IProjectService {
     }
 
     private boolean isValidProjectName(String name) {
-        return 3 <= name.length() && name.length() <= 20;
+        return 3 <= name.length() && name.length() <= 1024;
     }
 
     @Override

--- a/packages/sirius-web/frontend/sirius-web/src/views/edit-project/EditProjectNavbar/EditProjectNavbar.tsx
+++ b/packages/sirius-web/frontend/sirius-web/src/views/edit-project/EditProjectNavbar/EditProjectNavbar.tsx
@@ -48,6 +48,10 @@ const useEditProjectViewNavbarStyles = makeStyles((theme) => ({
   },
   title: {
     marginRight: theme.spacing(2),
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    maxWidth: '100ch',
   },
   onDarkBackground: {
     '&:hover': {

--- a/packages/sirius-web/frontend/sirius-web/src/views/new-project/NewProjectViewMachine.ts
+++ b/packages/sirius-web/frontend/sirius-web/src/views/new-project/NewProjectViewMachine.ts
@@ -63,7 +63,7 @@ export type NewProjectEvent =
   | HideToastEvent;
 
 const isChangeNameEvent = (event: NewProjectEvent): event is ChangeNameEvent => !!(event as ChangeNameEvent).name;
-const isNameInvalid = (name: string) => name.trim().length < 3 || name.trim().length > 20;
+const isNameInvalid = (name: string) => name.trim().length < 3 || name.trim().length > 1024;
 const isCreateProjectSuccessPayload = (payload: GQLCreateProjectPayload): payload is GQLCreateProjectSuccessPayload =>
   payload.__typename === 'CreateProjectSuccessPayload';
 export const newProjectViewMachine = Machine<NewProjectViewContext, NewProjectViewStateSchema, NewProjectEvent>(
@@ -72,7 +72,7 @@ export const newProjectViewMachine = Machine<NewProjectViewContext, NewProjectVi
     type: 'parallel',
     context: {
       name: '',
-      nameMessage: 'The name must contain between 3 and 20 characters',
+      nameMessage: 'The name must contain between 3 and 1024 characters',
       nameIsInvalid: false,
       message: null,
       newProjectId: null,

--- a/packages/sirius-web/frontend/sirius-web/src/views/projects/ProjectsView.tsx
+++ b/packages/sirius-web/frontend/sirius-web/src/views/projects/ProjectsView.tsx
@@ -461,7 +461,10 @@ const ProjectsTable = ({ projects, onMore }: ProjectsTableProps) => {
               <TableRow key={project.id} hover>
                 <TableCell>
                   <Link component={RouterLink} to={`/projects/${project.id}/edit`} color="inherit">
-                    {project.name}
+                    <div
+                      style={{ overflow: 'hidden', textOverflow: 'ellipsis', whiteSpace: 'nowrap', maxWidth: '100ch' }}>
+                      {project.name}
+                    </div>
                   </Link>
                 </TableCell>
                 <TableCell align="right">

--- a/packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItem.tsx
+++ b/packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItem.tsx
@@ -90,8 +90,12 @@ const useTreeItemStyle = makeStyles((theme) => ({
     gap: '4px',
     alignItems: 'center',
     cursor: 'pointer',
+  },
+  label: {
+    overflow: 'hidden',
     textOverflow: 'ellipsis',
     whiteSpace: 'nowrap',
+    maxWidth: '100ch',
   },
   selectedLabel: {
     fontWeight: 'bold',
@@ -323,7 +327,7 @@ export const TreeItem = ({
     );
   } else {
     text = (
-      <Typography variant="body2" className={selected ? classes.selectedLabel : ''}>
+      <Typography variant="body2" className={`${classes.label} ${selected ? classes.selectedLabel : ''}`}>
         {item.label}
       </Typography>
     );

--- a/scripts/restart-postgresql.sh
+++ b/scripts/restart-postgresql.sh
@@ -1,0 +1,3 @@
+#!/bin/sh
+docker kill sirius-web-postgres
+docker run -p 5433:5432 --name sirius-web-postgres -e POSTGRES_USER=dbuser -e POSTGRES_PASSWORD=dbpwd -e POSTGRES_DB=sirius-web-db -d postgres

